pkg/devicerepository/config.go

```diff
@@ -19,30 +19,45 @@ import (
 
 	"go.thethings.network/lorawan-stack/v3/pkg/config"
 	"go.thethings.network/lorawan-stack/v3/pkg/devicerepository/store"
+	"go.thethings.network/lorawan-stack/v3/pkg/devicerepository/store/bleve"
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
 )
 
 // Config represents the DeviceRepository configuration.
 type Config struct {
 	Store StoreConfig `name:"store"`
 
-	ConfigSource string                `name:"config-source" description:"Source of the Device Repository (static, directory, url, blob)"`
-	Static       map[string][]byte     `name:"-"`
-	Directory    string                `name:"directory" description:"OS filesystem directory, which contains Device Repository package"`
-	URL          string                `name:"url" description:"URL, which contains Device Repository package"`
-	Blob         config.BlobPathConfig `name:"blob"`
+	ConfigSource string `name:"config-source" description:"Source of the Device Repository (directory)"`
+	Directory    string `name:"directory" description:"OS filesystem directory, which contains Device Repository package"`
 
 	AssetsBaseURL string `name:"assets-base-url" description:"The base URL for assets"`
 }
 
 // StoreConfig represents configuration for the Device Repository store.
 type StoreConfig struct {
 	Store store.Store `name:"-"`
+
+	Bleve bleve.Config `name:"bleve"`
 }
 
 // NewStore creates a new Store for end devices.
 func (c Config) NewStore(ctx context.Context, blobConf config.BlobConfig) (store.Store, error) {
 	if c.Store.Store != nil {
 		return c.Store.Store, nil
 	}
-	return &store.NoopStore{}, nil
+
+	return c.Store.Bleve.NewStore(ctx)
+}
+
+var errFetcherConfig = errors.DefineInvalidArgument("fetcher_config", "only the directory configuration source is supported")
```

Comment:
This is going to backfire badly if we forget to update this message. Please harmonize with error messages of similar cases.